### DISCUSSION TITLE
Strong end-to-end encryption

### DISCUSSION THREAD

**Discussion by gluk64** [outline/outline#1783] [Ideas]
Huge number of people (myself included) would pay a large premium for end-to-end encryption of personal and team notes, instantly searchable offline on Desktop + Mobile.

This is the single most important feature that will bring you millions of users. No one else offers this now. The only two working exception are:

- Apple notes (encryption is very inconvenient and private notes not searchable).
- Standard notes (not working well offline with html).

Neither of these product supports teams.

**Comment by tommoor** [comment]
I would love to offer this functionality in the product, I agree that it would be a huge differentiator. Unfortunately as far as I'm aware the reason that "No one else offers this now" is because it's simply not possible – you can't have notes that are strongly encrypted _and_ have them searchable.

The compromise we offer for now is to allow anyone to host the software, you can then encrypt the disk however you like – and of course everything is encrypted in transit by default.
